lib/galaxy/schema/visualization.py

```diff
@@ -100,3 +108,251 @@ class VisualizationSummaryList(RootModel):
+    )
+    slug: Optional[str] = Field(
+        None,
+        title="Slug",
+        description="The slug of the visualization.",
+    )
+    latest_revision: VisualizationRevisionResponse = Field(
+        ...,
+        title="Latest Revision",
+        description="The latest revision of this Visualization.",
+    )
+    revisions: List[EncodedDatabaseIdField] = Field(
+        ...,
+        title="Revisions",
+        description="A list of encoded IDs of the revisions of this Visualization.",
+    )
+    url: str = Field(
+        ...,
+        title="URL",
+        description="The URL of the visualization.",
+    )
+    username: str = Field(
+        ...,
+        title="Username",
+        description="The name of the user owning this Visualization.",
+    )
+    email_hash: str = Field(
+        ...,
+        title="Email Hash",
+        description="The hash of the email of the user owning this Visualization.",
+    )
+    tags: Optional[TagCollection] = Field(
+        None,
+        title="Tags",
+        description="A list of tags to add to this item.",
+    )
+    annotation: Optional[str] = Field(
+        None,
+        title="Annotation",
+        description="The annotation of this Visualization.",
+    )
+    plugin: Optional[VisualizationPluginResponse] = Field(
+        None,
+        title="Plugin",
+        description="The plugin of this Visualization.",
+    )
+
+
+class VisualizationCreateResponse(Model):
+    id: EncodedDatabaseIdField = Field(
+        ...,
+        title="ID",
+        description="Encoded ID of the Visualization.",
+    )
+
+
+class VisualizationUpdateResponse(Model):
+    id: EncodedDatabaseIdField = Field(
+        ...,
+        title="ID",
+        description="Encoded ID of the Visualization.",
+    )
+    revision: EncodedDatabaseIdField = Field(
+        ...,
+        title="Revision",
+        description="Encoded ID of the Visualization Revision.",
+    )
+
+
+class VisualizationCreatePayload(Model):
+    type: Optional[SanitizedString] = Field(
+        None,
+        title="Type",
+        description="The type of the visualization.",
+    )
+    title: Optional[SanitizedString] = Field(
+        SanitizedString("Untitled Visualization"),
+        title="Title",
+        description="The name of the visualization.",
+    )
+    dbkey: Optional[SanitizedString] = Field(
+        None,
+        title="DbKey",
+        description="The database key of the visualization.",
+    )
+    slug: Optional[SanitizedString] = Field(
+        None,
+        title="Slug",
+        description="The slug of the visualization.",
+    )
+    annotation: Optional[SanitizedString] = Field(
+        None,
+        title="Annotation",
+        description="The annotation of the visualization.",
+    )
+    config: Optional[dict] = Field(
+        {},
+        title="Config",
+        description="The config of the visualization.",
+    )
+    save: Optional[bool] = Field(
```

Comment:
What is a visualization that is not saved ?

Reply:
I just added this because it existed in the old API!
But yes that is useless to have such a variable.

Reply:
I don't know ? Maybe that's used to preview a visualization ? But if so it would be broken since it doesn't validate against the response model.

Reply:
It is hardcoded from the old code to True!
https://github.com/galaxyproject/galaxy/blob/release_24.1/lib/galaxy/webapps/galaxy/api/visualizations.py#L304
It will remove it.